PlayCover/Views/App Views/StoreAppView.swift

```diff
@@ -67,33 +117,73 @@ struct StoreAppConditionalView: View {
                     .padding(.horizontal, 5)
                     .foregroundColor(.secondary)
             }
+            .onAppear {
+                getIconURLFromBundleIdentifier(app.id, app.region) { url in
+                    iconUrl = url
+                }
+            }
         } else {
             VStack(alignment: .center, spacing: 0) {
-                AsyncImage(
-                    url: URL(string: app.icon),
-                    content: { image in
+                VStack {
+                    AsyncImage(url: iconUrl) { image in
                         image
                             .resizable()
-                    },
-                    placeholder: {
+                            .aspectRatio(contentMode: .fit)
+                    } placeholder: {
                         ProgressView()
                             .progressViewStyle(.circular)
                     }
-                )
-                .frame(width: 88, height: 88)
-                .cornerRadius(10)
+                }
+                .cornerRadius(15)
+                .frame(width: 70, height: 70)
                 .shadow(radius: 1)
-                .padding(.top, 8)
+                .padding(.vertical, 5)
                 HStack {
                     Image(systemName: "arrow.down.circle")
                         .font(.system(size: 16))
                     Text(app.name)
                         .lineLimit(2)
                         .multilineTextAlignment(.center)
                 }
-                .padding(.bottom, 14)
+                .padding(.vertical, 5)
             }
             .frame(width: 150, height: 150)
+            .onAppear {
+                getIconURLFromBundleIdentifier(app.id, app.region) { url in
+                    iconUrl = url
+                }
+            }
+        }
+    }
+
+    public func getIconURLFromBundleIdentifier(_ bundleIdentifier: String,
+                                               _ region: StoreAppData.Region,
+                                               completion: @escaping (URL) -> Void) {
+        let url: URL
+
+        if region == .CN {
+            url = URL(string: "http://itunes.apple.com/lookup?bundleId=\(bundleIdentifier)" + "&country=cn")!
+        } else {
+            url = URL(string: "http://itunes.apple.com/lookup?bundleId=\(bundleIdentifier)")!
```

Comment:
If we don't specify country then it will query US store front. So all app not in US store will return empty.

Reply:
Currently, the only regions defined by the `region` enum are Global and CN, so the only regions that need to be checked are those two. If in future we change to include more region identifiers, this will need to be altered to reflect that.

Reply:
There are several other regions that need to be added as exemplified in https://ipa.playcover.workers.dev/1:/IPAs/

Reply:
Will handle in a different pr